src/sparql/illegal-date-violation.sparql

```diff
@@ -0,0 +1,12 @@
+PREFIX dct: <http://purl.org/dc/terms/>
+PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>
+PREFIX oboInOwl: <http://www.geneontology.org/formats/oboInOwl#>
+
+SELECT DISTINCT ?term ?property ?value WHERE
+{
+  VALUES ?property {dct:date dct:issued dct:created oboInOwl:creation_date}
+  ?term ?property ?value .
+  FILTER (datatype(?value) != xsd:date || !regex(str(?value), '^\\d{4}-\\d\\d-\\d\\d$'))
```

Comment:
I think the logic is slightly wrong, check this:

```suggestion
  FILTER (
    ( 
       datatype(?value) != xsd:date && !regex(str(?value), '^\\d{4}-\\d\\d-\\d\\d$')
     ) || ( 
       datatype(?value) != xsd:dateTime && !regex(str(?value), '^\\d{4}-\\d{2}-\\d{2}T\\d{2}:\\d{2}:\\d{2}Z')
     )
   )
```

Either the value is an xsd:date with the respective format, or it is an xsd:dateTime with the other format.

I could be wrong, and I did not check the syntax..

Reply:
If so, it needs to change in ODK.

Reply:
I trust your reading, what do you think?

Reply:
I tested your suggestion, and it returns all the correct cases. Having two filters for the same value is a way to have OR.

Reply:
> Having two filters for the same value is a way to have OR.

I did not know that.. Completely shocked!

Reply:
It's working as expected, so I'll leave like this.